packages/core/src/resolveTrustChains/policies/applyMetadataPolicyToMetadata.ts

```
@@ -0,0 +1,153 @@
+import { objectToEntries } from '../../utils/data'
+
+import { cloneDeep } from '../../utils/data'
+
+import {
+  type Metadata,
+  type MetadataPolicyOperator,
+  type SupportedPolicyKey,
+  allSupportedPolicies,
+  isExistingPolicyKey,
+} from '../../metadata'
+import { MetadataHelper } from './MetadataHelper'
+import { PolicyValidationError } from './errors/PolicyValidationError'
+import { type PolicyValue, union } from './utils'
+
+export async function applyMetadataPolicyToMetadata({
+  leafMetadata,
+  policyMetadata,
+}: { leafMetadata: Metadata; policyMetadata: Record<string, Record<string, MetadataPolicyOperator>> }) {
+  const resolvedLeafMetadata = new MetadataHelper(cloneDeep(leafMetadata))
+
+  for (const [serviceKey, service] of objectToEntries(policyMetadata)) {
+    for (const [servicePropertyKey, policyValue] of objectToEntries(service)) {
+      const policies = objectToEntries(policyValue)
+        .filter(([key]) => isExistingPolicyKey(key))
```

If a policy is unsupported, should this be silently ignored? I guess it should fail immediately in this case?

---

when it's unsupported it depends if it's included in the metadata_crit. If it's not in there it's allowed to be skipped. the`metadata_policy_crit` is being used in the resolution so at every level of the federation. 

When a `metadata_policy_crit` is present and the function is being used in the policy and it's not supported then it will fail that specific chain.